### PR TITLE
onnx.symbolic_helper.parse_args: document and clean up

### PR DESCRIPTION
**Pull request by garymm**
Fixes #{issue number}

### CODE REVIEW

File: torch/onnx/symbolic_helper.py [hunk 1]
```diff
@@ -126,9 +126,33 @@ def _is_packed_list(list_value):
 
 
 def parse_args(*arg_descriptors):
+    """A decorator which converts args from torch._C.Value to built-in types.
```
Comment (spandantiwari) [inline torch/onnx/symbolic_helper.py:129]: Great!

**Review by spandantiwari — APPROVED** [review]
LGTM.

**Comment by garymm** [comment]
@spandantiwari do the test failures seem related to this change?
I tried re-running locally and the test passed:
`python ./test/test_autograd.py TestAutograd.test_profiler_propagation`

And I'm not sure how to re-run the backward compatibility test locally.

File: torch/onnx/symbolic_helper.py [hunk 2]
```diff
@@ -146,11 +170,7 @@ def wrapper(g, *args, **kwargs):
             if len(kwargs) == 1:
                 assert '_outputs' in kwargs
             return fn(g, *args, **kwargs)
-        # In Python 2 functools.wraps chokes on partially applied functions, so we need this as a workaround
```
Comment (BowenBao) [inline torch/onnx/symbolic_helper.py:149]: (Given the previous comment..) Is this change ok with Python 2? CI does not cover ONNX export under python 2.

Comment (garymm) [reply torch/onnx/symbolic_helper.py:149]: According to this, PyTorch dropped support for Python 2 ~1.5 years ago: https://discuss.pytorch.org/t/dropping-python-2-support-on-january-1st-2020/56730

Is there some reason the code in torch/onnx still needs to support Python 2 when the rest of torch does not?

Comment (BowenBao) [reply torch/onnx/symbolic_helper.py:149]: i see, thanks for updating this part.

**Pull request merged**